src/apple/processor.rs

```diff
@@ -229,6 +229,9 @@ fn get_sysctl_str(s: &[u8]) -> String {
     if len > 0 {
         unsafe {
             buf.set_len(len);
+            if buf[len - 1] == b'\0' {
```

Comment:
This fix seems not great: what happens if there are in fact more than one `\0`? Please take this into account too. :)

Reply:
Fair enough! I assumed it was just the C terminator not getting handled, but the new code is more robust.